### PR TITLE
First changes to compat with rust 1.0.0-dev 1fd89b625 2015-04-07

### PR DESCRIPTION
**Pull request by paulosuzart**
Hi,

this makes this project compatible with rust beta, but compiled with a nightly version after the beta release. So, it is just a matter of time to do a double check with the final 1.0 release.

On running tests you'll notice something I'll explain in a moment.

bests

### CODE REVIEW

File: src/connection.rs [hunk 1]
```diff
@@ -47,70 +49,75 @@ impl Connection {
...
         };
         let length: usize = match header[3].trim().parse() {
-            Some(length) => { length }
-            None => { return Err(MemcacheError::ServerError); }
+            Ok(length) => { length }
+            Err(err) => { return Err(MemcacheError::ServerError); }
         };
-        let value = try!{ self.stream.read_exact(length) };
-        return Ok(Some((value, flags)));
+        let mut buf : Vec<u8> = Vec::with_capacity(length);
```
Comment (paulosuzart) [inline src/connection.rs]: This is returning an empty `Vec` even consuming the stream. I'm trying to work around in a separate branch.

**Comment by aisk** [comment]
Hi thanks for PR, I think we should keep master branch always available, so I will merge this after all test case passed.

**Comment by paulosuzart** [comment]
I see. In fact your current master isn't compatible with current rust version. But I got you. Do you think you can take a look at my branch and figure out what is preventing that commented line on PR to work properly?

**Comment by paulosuzart** [comment]
BTW you also have the option to tag the current master in a tag name 0.0.5 and point crates.io to it. Then you can pull this request and continue with improvements. It is up to  you. I'll try harder to find out what is happening to this test failing.

File: src/connection.rs [hunk 2]
```diff
@@ -47,70 +49,79 @@ impl Connection {
...
-            Some(length) => { length }
-            None => { return Err(MemcacheError::ServerError); }
+            Ok(length) => { length }
+            Err(_) => { return Err(MemcacheError::ServerError); }
         };
-        let value = try!{ self.stream.read_exact(length) };
-        return Ok(Some((value, flags)));
+        let mut buf : Vec<u8> = Vec::with_capacity(length);
+        //buf.resize(length, 0); Safe option to the code bellow
+        unsafe {
```
Comment (paulosuzart) [inline src/connection.rs]: Solved with unsafe operation. But safe enough provided that the length is known at this time.

**Comment by paulosuzart** [comment]
Now all tests ok!

**Comment by aisk** [comment]
Cool, I will merge this, and release 0.0.6.

**Pull request merged**